crates/rune/src/languageserver/state.rs

```diff
@@ -278,6 +480,12 @@ impl<'a> State<'a> {
 
                 source.index = value;
                 source.build_sources = Some(sources.clone());
+
+                if let Ok(unit) = unit.as_ref().map(|v| v.clone()) {
+                    source.unit = Some(unit.clone());
+                }
+
+                source.docs = Some(doc_visitor.clone());
```

Comment:
This is definitely wrong; but there seems to be some missing "level" in between here. (I.e, what's the build target mapping on a workspace?)

Reply:
Having to clone out the docs to multiple sources feels wrong out to me.